### PR TITLE
[DTensor] Turn on foreach implementation for clip_grad_norm_ for DTensor by default 

### PR DESCRIPTION
**Pull request by wz337**
Fixes #ISSUE_NUMBER

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @tianyu-l @wconstab @yf225 @chauhang @d4l3k @msaroufim

### CODE REVIEW

File: test/distributed/_composable/fsdp/test_fully_shard_clip_grad_norm_.py [hunk 1]
```diff
@@ -67,11 +67,11 @@ def _test_clip_grad_norm(
             )
             comm_mode = CommDebugMode()
             with comm_mode:
+                # foreach is default to turn on so we don't need to specify it.
```
Comment (awgu) [inline test/distributed/_composable/fsdp/test_fully_shard_clip_grad_norm_.py:70]: nit: haha I would be okay to just not include this comment

**Review by awgu — APPROVED** [review]
Looks great!

File: test/distributed/_composable/fsdp/test_fully_shard_clip_grad_norm_.py (hunk 1 above)
Comment (wz337) [reply test/distributed/_composable/fsdp/test_fully_shard_clip_grad_norm_.py:70]: Sure. You are so fast! Looks like the `test_clip_grad_norm_2d` errors out on main T_T and thereby erroring out on the PR as well. Looking now.

**Pull request closed without merging**